### PR TITLE
GetValidatorPerformance only return its status info if the validator is active

### LINKED ISSUE #5485: Validators with PENDING status included in Previous epoch voting summary
image

### PR DESCRIPTION
**Pull request by ShawkiS**
### Resolves #5485

**Description
Write why you are making the changes in this pull request:**
The problem here is that the user has 5 validator keys. Only 2 of these are active. The feature request is to only log "previous epoch voting summary" for activated validators as this log is not helpful for pending ones. A pending validator had no duties in the last epoch and there is really nothing to report on their performance.

**Write a summary of the changes you are making:**
This PR changes one file that is `rpc/beacon/validators.go` and it adds a condition to the GetValidatorPerformance to make sure it only returns its status info if the validator is active

### CODE REVIEW

**Comment by prestonvanloon** [comment]
Please fix build
```

/root/.cache/bazel/_bazel_root/5c04516d923b21f838ef18b7f8ed7f88/execroot/prysm/beacon-chain/rpc/beacon/validators.go:817:16: undefined: errors
--
  | /root/.cache/bazel/_bazel_root/5c04516d923b21f838ef18b7f8ed7f88/execroot/prysm/beacon-chain/rpc/beacon/validators.go:819:32: not enough arguments in call to helpers.IsActiveValidator
  | have (*eth.Validator)
  | want (*eth.Validator, uint64)


```

File: beacon-chain/rpc/beacon/validators.go [hunk 1]
```diff
@@ -806,11 +806,21 @@ func (bs *Server) GetValidatorPerformance(
...
 	// Also track missing validators using public keys.
 	for _, key := range req.PublicKeys {
 		pubkeyBytes := bytesutil.ToBytes48(key)
 		idx, ok := headState.ValidatorIndexByPubkey(pubkeyBytes)
+		val, err := headState.ValidatorAtIndex(idx)
+		if err != nil {
+			return nil, errors.Wrap(err, "could not get validator")
+		}
+		if !helpers.IsActiveValidator(val) {
+		// If the validator isn't active, no need to get its performance
```
Comment (terencechain) [inline beacon-chain/rpc/beacon/validators.go:820]: Not sure if the comment here is necessary. It's pretty self explanatory imo

**Comment by terencechain** [comment]
Please also add a description to the PR

**Comment by ShawkiS** [comment]
@prestonvanloon Okay will fix it, 

I saw two exapmples: one passing currentEpoch as `helpers.SlotToEpoch(currentSlot)` and the other `helpers.PrevEpoch(state)` 

What shall I pass in our current example?

@terencechain okay will remove it

File: beacon-chain/rpc/beacon/validators.go [hunk 2]
```diff
@@ -802,15 +802,25 @@ func (bs *Server) GetValidatorPerformance(
 	correctlyVotedHead := make([]bool, 0, reqPubKeysCount)
 	missingValidators := make([][]byte, 0, reqPubKeysCount)
 
-	headState, err := bs.HeadFetcher.HeadState(ctx)
-	if err != nil {
-		return nil, status.Error(codes.Internal, "Could not get head state")
-	}
+    headState, err := bs.HeadFetcher.HeadState(ctx)
```
Comment (nisdas) [inline beacon-chain/rpc/beacon/validators.go:805]: please use `gofmt` here the indentation is off.

File: beacon-chain/rpc/beacon/validators.go [hunk 3]
```diff
@@ -802,15 +802,25 @@ func (bs *Server) GetValidatorPerformance(
...
+ 
 	// Convert the list of validator public keys to list of validator indices.
 	// Also track missing validators using public keys.
 	for _, key := range req.PublicKeys {
 		pubkeyBytes := bytesutil.ToBytes48(key)
 		idx, ok := headState.ValidatorIndexByPubkey(pubkeyBytes)
+		val, err := headState.ValidatorAtIndex(idx)
+		currentEpoch := helpers.CurrentEpoch(headState)
+		if err != nil {
+			return nil, status.Errorf(codes.Internal, "could not get validator: %v", err)
```
Comment (nisdas) [inline beacon-chain/rpc/beacon/validators.go:818]: fix the error here 
```
could not get current epoch
```

File: beacon-chain/rpc/beacon/validators.go (hunk 2 above)
Comment (ShawkiS) [reply beacon-chain/rpc/beacon/validators.go:805]: Okay

File: beacon-chain/rpc/beacon/validators.go (hunk 3 above)
Comment (ShawkiS) [reply beacon-chain/rpc/beacon/validators.go:818]: Oh, why that is happening I saw this code working in other places?
Is the way I am getting it wrong?

Comment (nisdas) [reply beacon-chain/rpc/beacon/validators.go:818]: ok so your order of code here is vulnerable to errors here. if you look at 
```go
		idx, ok := headState.ValidatorIndexByPubkey(pubkeyBytes)
```
you need to handle the case where the index doesn't exist. 
so immediately after that you need to do:
```go
		if !ok {
			missingValidators = append(missingValidators, key)
			continue
		}
```
Then  doing this: 
```go
		val, err := headState.ValidatorAtIndex(idx)
		currentEpoch := helpers.CurrentEpoch(headState)
		if err != nil {
```
is wrong you need to handle the error first 
```go
		val, err := headState.ValidatorAtIndex(idx)
		if err != nil {
                        return nil, status.Errorf(codes.Internal, "could not get validator: %v", err)
                }		
               currentEpoch := helpers.CurrentEpoch(headState)
```

File: beacon-chain/rpc/beacon/validators.go [hunk 4]
```diff
@@ -804,8 +804,9 @@ func (bs *Server) GetValidatorPerformance(
 
 	headState, err := bs.HeadFetcher.HeadState(ctx)
 	if err != nil {
-		return nil, status.Error(codes.Internal, "Could not get head state")
+		return nil, status.Errorf(codes.Internal, "could not get head state: %v", err)
```
Comment (nisdas) [inline beacon-chain/rpc/beacon/validators.go:807]:
```suggestion
		return nil, status.Errorf(codes.Internal, "Could not get head state: %v", err)
```
This is a server error returned to our rpc clients, so it should be capitalized.

**Comment by rauljordan** [comment]
Hi @ShawkiS would also recommend always adding a PR description and a good title that gives us enough context about why this is needed (if it solves an open issue, type `resolves #ISSUE_NUM` or mention the issue it is related to in the description). Also use labels as needed, such as when the PR is ready for review or when it is still in progress.

**Comment by ShawkiS** [comment]
Hey @rauljordan, okay will make sure to do that next time because it seems that I can't edit the description now, yeah that's better I changed the name. Does it seem that I can't set a label?

**Pull request merged**